dot/sync/chain_processor.go

```diff
@@ -70,11 +70,11 @@ func (s *chainProcessor) stop() {
 func (s *chainProcessor) processReadyBlocks() {
 	for {
 		bd := s.readyBlocks.pop(s.ctx)
-		if s.ctx.Err() != nil {
+		if bd == nil { // context was canceled
```

Comment:
not necessarily a helpful comment for your requested review but the code would be more readable if we did not use variable names like `s` or `bd`. 

```go
chainProcessorContext := chainProcessor.readyBlocks.pop(chainProcessor.ctx)
```

Reply:
also, inline comments should be a separate line above, what do you think?

```
// check if the chain-processor context was canceled
if chainProcessorContext == nil {
```

Reply:
- for method receivers, I made peace with having short or single letter variable names. Although here it should be `c`/`cp`, not `s` (historical reasons I guess).
- for all other variables, short non-explicit names are a big no-no for me. Completely agree. At least for new code we add/modify I think our team now enforces explicit variable names.

For both cases, as you mentioned, it's a bit out of scope to keep deltas somehow low in this PR.

> also, inline comments should be a separate line above, what do you think?

For me, it really depends, if the comment is short and relates to a single line (here `if bd == nil {`), it's fine to inline it. If the comment is long or refers to a block of code below, then yeah have it on a separate line.
Anyway, I think it's better to leave this up to the developer, it's not a big deal and we shouldn't enforce too much code style or contributors (us internally or external) would just get mad :exploding_head: :wink:

Reply:
Yeah I agree with you too Tim, reverting the code to how it was is also confusing I think.

I did tentative commit 015c45d6425e44592c2b1db014fdaa6bd7e3f4b9 changing `pop` to return an error when the context gets canceled? That way it would be more explicit, what do you think? Or something else